pkg/controller/kieapp/kieapp_controller.go

```diff
@@ -966,6 +930,36 @@ func (reconciler *Reconciler) getCSV(operator *appsv1.Deployment) *operatorsv1al
 	return csv
 }
 
+func (reconciler *Reconciler) addConsoleLinkResources(resources []resource.KubernetesResource, cr *api.KieApp) []resource.KubernetesResource {
+	if cr.Status.ConsoleHost == "" || !strings.HasPrefix(cr.Status.ConsoleHost, "https://") {
+		return resources
+	}
+	consoleLink := &consolev1.ConsoleLink{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				"rhpam-namespace": cr.Namespace,
+				"rhpam-app":       cr.Name,
+			},
+		},
+		Spec: consolev1.ConsoleLinkSpec{
+			Link: consolev1.Link{
+				Href: cr.Status.ConsoleHost,
+				Text: constants.EnvironmentConstants[cr.Spec.Environment].App.FriendlyName,
+			},
+			Location: consolev1.NamespaceDashboard,
+			NamespaceDashboard: &consolev1.NamespaceDashboardSpec{
+				Namespaces: []string{cr.Namespace},
+			},
+		},
+	}
+	if cr.Status.ConsoleLink != "" {
+		consoleLink.ObjectMeta.Name = cr.Status.ConsoleLink
+	}
+	consoleLink.ObjectMeta.GenerateName = fmt.Sprintf("%s-%s-", cr.Namespace, cr.Name)
+	controllerutil.AddFinalizer(cr, constants.ConsoleLinkFinalizer)
```

Comment:
If required, should be deleted in line 165

Reply:
the finalizer is deleted by the extendedReconciler

Reply:
No, I'm saying that I don't see the need for a finalizer. That line 165 of the controller will delete `ConsoleLink` as needed.

Reply:
The finalizer string has to be added to the kieapp resource to specify that there is something that needs to be deleted.
During deletion, the extended reconciler will go through the kieapp finalizers array and look for the registered Finalizers, call the OnFinalizeFn and finally remove the finalizer string from the kieapp resource.

Reply:
You're thinking about your solution... forget about finalizers for a second and just look at the reconcile loop. It already deletes whatever shouldn't be there.

Reply:
Ok, I see. The fact that the comment was in this line blinded me